src/components/titlePanel/createTitlePanel.ts

```diff
@@ -0,0 +1,90 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+import { DNode, Widget, WidgetFactory, WidgetProperties } from '@dojo/widget-core/interfaces';
+import themeable, { ThemeableMixin } from '@dojo/widget-core/mixins/themeable';
+
+import * as css from './styles/titlePanel.css';
+import * as animations from '../../styles/animations.css';
+
+export interface TitlePanelProperties extends WidgetProperties {
+	collapsed?: boolean;
+	collapsible?: boolean;
+	enterAnimation?: string;
+	exitAnimation?: string;
+	title: string;
+
+	onRequestCollapse?(): void;
+	onRequestExpand?(): void;
+};
+
+export type TitlePanel = Widget<TitlePanelProperties> & ThemeableMixin & {
+	onClickTitle(): void;
+};
+
+export interface TitlePanelFactory extends WidgetFactory<TitlePanel, TitlePanelProperties> { };
+
+const createTitlePanel: TitlePanelFactory = createWidgetBase
+	.mixin(themeable)
+	.mixin({
+		mixin: {
+			baseClasses: css,
+
+			onClickTitle(this: TitlePanel) {
+				const {
```

Comment:
Mangala, you don't have to use noop defaults if you don't want to (I didn't in SlidePane), but I'd still destructure fully to avoid the four extra uses of `this.properties`.